packages/jest-mock/src/index.ts

```diff
@@ -1082,34 +1090,34 @@ export class ModuleMocker {
         const originalGet = descriptor.get;
         mock = this._makeComponent({type: 'function'}, () => {
           descriptor!.get = originalGet;
-          Object.defineProperty(object, methodName, descriptor!);
+          Object.defineProperty(object, methodKey, descriptor!);
         });
         descriptor.get = () => mock;
-        Object.defineProperty(object, methodName, descriptor);
+        Object.defineProperty(object, methodKey, descriptor);
       } else {
         mock = this._makeComponent({type: 'function'}, () => {
           if (isMethodOwner) {
-            object[methodName] = original;
+            object[methodKey] = original;
           } else {
-            delete object[methodName];
+            delete object[methodKey];
           }
         });
         // @ts-expect-error overriding original method with a Mock
-        object[methodName] = mock;
+        object[methodKey] = mock;
       }
 
       mock.mockImplementation(function (this: unknown) {
         return original.apply(this, arguments);
       });
     }
 
-    return object[methodName];
+    return object[methodKey];
   }
 
-  private _spyOnProperty<T extends object, M extends PropertyLikeKeys<T>>(
+  private _spyOnProperty<T extends object, K extends PropertyLikeKeys<T>>(
     obj: T,
-    propertyName: M,
-    accessType: 'get' | 'set' = 'get',
```

Comment:
Default is unnecessary, `accessType` is always passed:

https://github.com/facebook/jest/blob/052aa67f22834933d986c8fb2a373379408ae5ea/packages/jest-mock/src/index.ts#L1043-L1045